src/core/data/transform.cc

```diff
@@ -239,6 +248,17 @@ void Project::print(std::ostream& out) const
 
 int32_t Project::target_ndim(int32_t source_ndim) const { return source_ndim + 1; }
 
+void Project::return_promoted_dims(std::vector<int32_t>& dims) const
+{
+  auto finder = std::find(dims.begin(), dims.end(), (dim_));
+  if (finder != dims.end()) {
+    dims.erase(finder);
+    for (size_t i = 0; i < dims.size(); i++) {
+      if (dims[i] > dim_) { dims[i] -= 1; }
+    }
```

Comment:
```suggestion
    for (auto& dim : dims)
      if (dim > dim_) --dim;
```

Reply:
also, don't you need to do this even when `dim_` wasn't found in the `dims`?

Reply:
wouldn't this function return `{1}` for `store.promote(1, 10).project(0, 0).project(0, 0)`?

Reply:
Yes, we need to do `--dim` even if `dim_` is not found in `dims`